### PR TITLE
support custom 404 (static html only)

### PR DESCRIPTION
**Pull request by dai-shi**
Jut put `./public/404.html`.

### CODE REVIEW

**Comment by dai-shi** [comment]
@aheissenberger Can you work on AWS Lambada?
@threepointone Can you work on Partykit?

If this PR is still open, send a PR to this branch, otherwise to `main`.

File: examples/01_template/public/404.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<h1>Custom Not Found Page</h1>
```
Comment (himself65) [inline examples/01_template/public/404.html:1]: What if I want use javascript here?

Comment (himself65) [reply examples/01_template/public/404.html:1]: This looks like a manual setup not related to Waku itself

Comment (dai-shi) [reply examples/01_template/public/404.html:1]: Think it like a static image asset file. Up until now, it's very hard coded. So, this at least makes it configurable. And, it works without SSR.

I plan to do full customization in #329. But, if we want a solution "with React" we need another solution. I'll think about it, but it requires SSR.

A question for you: How should we deal with customizing 404 without SSR? What's the typical way with Vite + React SPA?

> not related to Waku itself

Yeah, it's not related to RSC. But, we need do something and this is a minimal setup.

**Pull request merged**